44705: Add clarity to updating-existing-cluster.md (#230)
replicatedhq/replicated-docs: docs/enterprise/updating-existing-cluster.md

# Updating the Admin Console on an Existing Cluster

This topic explains how to upgrade the Replicated admin console on an existing cluster.
For information about how to upgrade the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer cluster](updating-embedded-cluster).

Updating the admin console requires upgrading the kots CLI KOTS first. You have the option to upgrade to the latest version of the kots CLI or to a specific earlier version. Then, the admin console version that you upgrade to will match the kots CLI version.

### Update an Online Installations

To update an online installation of the admin console:

> 1. Run the `kubectl kots version` to download the latest binary.

Author: Done! I'm marking it as an optional step. Also, I was confused by that text, which was in the original. It didn't seem like it was gonna download anything. Thanks!

1. Do _one_ of the following actions to update your kots CLI version:

    - For the latest version, run `curl https://kots.io/install | bash`.

    - For a particular version, run `curl https://kots.io/install/<version> | bash`.

1. Run the following command to update the admin console on an existing cluster:

  ```bash
  kubectl kots admin-console upgrade -n <namespace>
  ```

  :::note
  For help information, run `kubectl kots admin-console upgrade -h`.
  :::

### Update an Air Gap Installation

To update an air gap installation of the admin console:

> 1. Download the the latest Replicated app manager binary from [Github](https://github.com/replicatedhq/kots/releases) or the customer download page in the Replicated [vendor portal](https://vendor.replicated.com).

Author: Right, exactly. This was in the original topic as well and I didn't know why. Would it be correct to say "Download the latest Replicated app manager binary file from your vendor's customer download page."?

> 1. Download the the latest Replicated app manager binary from [Github](https://github.com/replicatedhq/kots/releases) or the customer download page in the Replicated [vendor portal](https://vendor.replicated.com).

Author: AH, I guess the Github part is ok, so I went with this: "Download your desired version of the Replicated app manager air gap bundle from [Github](https://github.com/replicatedhq/kots/releases) or from the customer download page provided by your vendor. The air gap bundle is named `kotsadm.tar.gz`."

> 1. Do _one_ of the following actions to update your kots CLI version:

Author: Added this as an Important admonition to catch their eye.

# Updating the Admin Console on an Existing Cluster

This topic explains how to upgrade the Replicated admin console on an existing cluster.
For information about how to upgrade the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer cluster](updating-embedded-cluster).

> Updating the admin console requires upgrading the kots CLI KOTS first. You have the option to upgrade to the latest version of the kots CLI or to a specific earlier version. Then, the admin console version that you upgrade to will match the kots CLI version.

Reviewer: Not necessarily requesting a change, but a musing I'm voicing aloud. The last sentence here feels more true for online installs than for air gap installs. The admin console version matches the kots CLI version for online installs because the kots CLI will install that version. For air gap installs, it matches because you have to make it match, by downloading matching CLI and air gap bundle versions. So this last sentence, especially the "then" preamble, feels true for online installs but not necessarily for air gap installs. Does that make sense?

It might be clearer to say what I have in the suggestion here, but let me know your thoughts.
```suggestion
Updating the admin console requires upgrading the kots CLI first. You have the option to upgrade to the latest version of the kots CLI or to a specific earlier version. For online installations, the admin console will be upgraded to the same version as the version of the kots CLI that is used to perform the upgrade. For air gap installations, the version of the kots CLI that is used to perform the upgrade must match the version of the KOTS air gap bundle used for the upgrade.
```

# Updating the Admin Console on an Existing Cluster

This topic explains how to update the Replicated admin console on an existing cluster.

> For information about how to update the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer cluster](updating-embedded-cluster).

Reviewer: Last suggestion! Based on our Slack convo.
```suggestion
For information about how to update the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer-created cluster](updating-embedded-cluster).
```

> For information about how to updating the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer cluster](updating-embedded-cluster).

Reviewer: ```suggestion
For information about how to update the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer cluster](updating-embedded-cluster).
```

> For information about how to updating the admin console on a Kubernetes installer-created cluster (embedded cluster), see [Updating the admin console on a Kubernetes installer cluster](updating-embedded-cluster).

Reviewer: I am also curious about "Kubernetes installer cluster" vs. "Kubernetes installer-created cluster" vs. "embedded cluster". Obviously embedded cluster was our original wording, but I see "Kubernetes installer cluster" a lot in these new docs. What is our official naming for that?

Author: Ah, yes, very confusing. I'll send you a link to our internal rebranding, created by Tracie.

> Updating the admin console requires updating the kots CLI first. You have the option to update to the latest version of the kots CLI or to a specific version.

Reviewer: @jonquil2002 Here's a suggestion on not allowing downgrades.
```suggestion
Updating the admin console requires updating the kots CLI first. You have the option to update to the latest version of the kots CLI or to a specific version.

Downgrading the admin console to a version earlier than what is currently deployed is not supported.
```

> ### Update an Online Installations

Reviewer: ```suggestion
### Update an Online Installation
```

> For online installations, the admin console is updated to the same version as the version of the kots CLI that is used to perform the update.

Reviewer 2: Two suggestions:
- "that is used to perform.." --> "that you use to perform..."
- Just since this concept is slightly more challenging to convey, recommend an example here. Maybe something like "For example, if you use v1.56.0 of the kots CLI to upgrade the admin console, then the admin console is upgraded to v1.56.0." Will leave it up to your discretion if this actually aids clarity, but just something to consider :)

To update an online installation of the admin console:

1. (Optional) Run the `kubectl kots version` command to see the current version of the kots CLI.

1. Do _one_ of the following actions to update your kots CLI version:

    - For the latest version, run `curl https://kots.io/install | bash`.

    - For a particular version, run `curl https://kots.io/install/<version> | bash`.

1. Run the following command to update the admin console on an existing cluster:

  ```bash
  kubectl kots admin-console upgrade -n <namespace>
  ```

  :::note
  For help information, run `kubectl kots admin-console upgrade -h`.
  :::

### Update an Air Gap Installation

> For air gap installations, the version of the kots CLI that is used to perform the update must match the version of the KOTS air gap bundle used for the update.

Reviewer 2: I think this instance of KOTS air gap bundle could be changed to Replicated app manager air gap bundle. It looks like on line 38 you do use app manager instead of KOTS when referring to the bundle. If you agree and decide to make this change, then on line 38, you could remove the "Replicated" before "app manager"

Author: Good catch!
